staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go

```diff
@@ -238,43 +306,242 @@ func newAuthenticator(opts Options, initVerifier func(ctx context.Context, a *Au
 		Now:                  now,
 	}
 
+	// TODO: Distributed claims resolution will be enabled unconditionally once
+	// response caching is implemented.
+	var resolver *claimResolver
+	if opts.enableDistributedClaims {
+		resolver = newClaimResolver(opts.GroupsClaim, client, verifierConfig)
+	}
+
+	authenticator := &Authenticator{
+		issuerURL:      opts.IssuerURL,
+		usernameClaim:  opts.UsernameClaim,
+		usernamePrefix: opts.UsernamePrefix,
+		groupsClaim:    opts.GroupsClaim,
+		groupsPrefix:   opts.GroupsPrefix,
+		requiredClaims: opts.RequiredClaims,
+		cancel:         cancel,
+		resolver:       resolver,
+	}
+
 	initVerifier(ctx, authenticator, verifierConfig)
 	return authenticator, nil
 }
 
-func hasCorrectIssuer(iss, tokenData string) bool {
-	parts := strings.Split(tokenData, ".")
+// untrustedIssuer extracts an untrusted "iss" claim from the given JWT token,
+// or returns an error if the token can not be parsed.  Since the JWT is not
+// verified, the returned issuer should not be trusted.
+func untrustedIssuer(token string) (string, error) {
+	parts := strings.Split(token, ".")
 	if len(parts) != 3 {
-		return false
+		return "", fmt.Errorf("malformed token %q", token)
```

Comment:
Don't include the token. It's a credential that shouldn't end up in logs or error messages